src/main/scala/io/iohk/ethereum/jsonrpc/server/http/JsonRpcHttpServer.scala

```diff
@@ -83,6 +83,15 @@ trait JsonRpcHttpServer extends Json4sSupport with RateLimit with Logger {
     }
   }
 
+  private def handleResponse(f: Task[JsonRpcResponse]): Task[(StatusCode, JsonRpcResponse)] = f map { jsonRpcResponse =>
+    jsonRpcResponse.error match {
+      case Some(JsonRpcError(-32600, _, _)) => (StatusCodes.BadRequest, jsonRpcResponse)
+      case Some(JsonRpcError(-32601, _, _)) => (StatusCodes.NotFound, jsonRpcResponse)
+      case Some(JsonRpcError(_, _, _)) => (StatusCodes.InternalServerError, jsonRpcResponse)
+      case None => (StatusCodes.OK, jsonRpcResponse)
+    }
+  }
+
```

Comment:
can we have a test case for every code specified in [3.6.2   Errors](https://www.jsonrpc.org/historical/json-rpc-over-http.html#id19)